src/postgraphile-upsert.ts

```diff
@@ -223,25 +223,104 @@ export const PgMutationUpsertPlugin: Plugin = (builder) => {
                     );
 
                   // Figure out to which columns the unique constraints belong to
-                  const uniqueKeys = uniqueConstraints.reduce(
-                    (acc, constraint) => [
-                      ...acc,
-                      ...constraint.keyAttributeNums.map((num) =>
-                        attributes.find((attr) => attr.num === num)
-                      ),
-                    ],
-                    []
+                  const constraintColumns: { [key: string]: any[] } =
+                    uniqueConstraints.reduce(
+                      (acc, constraint) => ({
+                        ...acc,
+                        [constraint.name]: new Set(
+                          constraint.keyAttributeNums.map((num) =>
+                            attributes.find((attr) => attr.num === num)
+                          )
+                        ),
+                      }),
+                      {}
+                    );
+
+                  const upsertedColumns: Set<string> = new Set(
+                    attributes.filter((attr) => {
+                      if (
+                        Object.prototype.hasOwnProperty.call(
+                          inputData,
+                          fieldName
```

Comment:
will this condition ever be true?

`{  model: "kula deluxe v2" weight: 0.0, serialKey: "1234" }` has fieldName `upsertBike`?